main/src/main/java/org/mobilitydata/gtfsvalidator/util/GeospatialUtil.java

```diff
@@ -66,98 +66,101 @@ private static DistanceCalculator getDistanceCalculator() {
       double fromLat, double fromLng, double toLat, double toLng) {
     final ShapeFactory shapeFactory = getShapeFactory();
     final DistanceCalculator distanceCalculator = getDistanceCalculator();
     final Point origin = shapeFactory.pointXY(fromLng, fromLat);
     final Point destination = shapeFactory.pointXY(toLng, toLat);
-    return (int)
-        (DistanceUtils.DEG_TO_KM
+    return DistanceUtils.DEG_TO_KM
             * distanceCalculator.distance(origin, destination)
-            * KILOMETER_TO_METER_CONVERSION_FACTOR);
+            * KILOMETER_TO_METER_CONVERSION_FACTOR;
   }
 
   /**
-   * Returns a list of E052 errors for the given input, one for each stop that is too far from the
+   * Returns a list of notices for the given input, one for each stop that is too far from the
    * trip shapePoints
    *
-   * @param trip Trip for this GTFS trip
-   * @param stopTimes a map of StopTimes for a trip, sorted by stop_sequence
-   * @param shapePoints a map of ShapePoints for a trip, sorted by shape_pt_sequence
-   * @param stopTable a map of all stops (keyed on stop_id), needed to obtain the latitude and
-   *     longitude for each stop
+   * @param tripId trip_id for this GTFS trip
+   * @param stopTimes a list of StopTimes for a trip, sorted by stop_sequence
+   * @param shapeId the shape_id for this GTFS trip
+   * @param shapePoints a list of ShapePoints for a trip, sorted by shape_pt_sequence
+   * @param stopTable a container for all stops (keyed on stop_id), needed to obtain the latitude and
+   *     longitude for each stop on the trip
    * @param testedCache a cache for previously tested shape_id and stop_id pairs (keyed on
    *     shape_id+stop_id). If the combination of shape_id and stop_id appears in this set, we
    *     shouldn't test it again. Shapes and stops tested in this method execution will be added to
    *     this testedCache.
-   * @return a list of E052 errors, one for each stop that is too far from the trip shapePoints
+   * @return a list of notices, one for each stop that is too far from the trip shape
    */
   public static List<StopTooFarFromTripShapeNotice> checkStopsWithinTripShape(
-      final GtfsTrip trip,
+      final String tripId,
       final List<GtfsStopTime> stopTimes,
+      final String shapeId,
       final List<GtfsShape> shapePoints,
       final GtfsStopTableContainer stopTable,
       final Set<String> testedCache) {
-    List<StopTooFarFromTripShapeNotice> errors = new ArrayList<>();
-    if (trip == null
-        || stopTimes == null
+    List<StopTooFarFromTripShapeNotice> notices = new ArrayList<>();
+    if (stopTimes == null
         || stopTimes.isEmpty()
         || shapePoints == null
         || shapePoints.isEmpty()) {
       // Nothing to do - return empty list
-      return errors;
+      return notices;
     }
 
     // Create a polyline from the GTFS shapes data
     ShapeFactory.LineStringBuilder lineBuilder = getShapeFactory().lineString();
```

Comment:
Good idea, but unfortunately no :(. I looked at the source of spatial4j, and it has a comment in the builder class:

>// TODO add dimensionality hint method?

Looks like it's backed by an ArrayList, so a lot of re-sizing will happen here :(

https://github.com/locationtech/spatial4j/blob/1f6e2047f0574a430fc711cf2cd5adf141a8bda9/src/main/java/org/locationtech/spatial4j/shape/jts/JtsShapeFactory.java#L312

Reply:
In good news, in looking at the library source I realized that you can specify a buffered line directly in the `lineBuilder`, without having to build the line first and then generating the buffer. So that saves memory and execution time for each trip shape.

However, I tested the change and initially got some confusing results. The unit test (using an original shape to line distance of around 140m - the threshold is 100m) passed with the change, but I ran it against MBTA's dataset with the old and new code, and unexpectedly got different results - directly buffering the line in the builder gave me 8 notices, while creating the second instance (original code) gave me 6 notices.

After more poking around in spatial4j code, it looks like the difference is that when buffering within `ShapeFactory.LineStringBuilder` directly, it always returns a non-geodesic (straight-line) buffer. Conversely, when you first build and then separately buffer the line, it always returns a geodesic buffer (i.e., one that follows the surface of the earth).

The differences between the two are slight, but enough to cause differences in results on real world datasets. I manually measured the distance between one of the stops and the shape in the newly generated notice using Google Earth Pro and it measures around 106 meters.

Another consideration - as you might expect, I noticed that execution time on Euclidean buffers seems to be consistently less than on geodesic buffers. Here are some measurements running on my laptop from within IntelliJ.

Execution time - Euclidean (seconds):
* 15
* 11
* 12
* 13
* 11
* 12

Execution time - Geodesic (seconds):
* 21
* 18
* 17
* 16
* 17
* 17
* 16

Given what seems to be a decent performance increase in execution time alone (memory performance should be better too, but I didn't measure it), I'm inclined to go with the Euclidean measurements. For our purposes I think the small error doesn't really matter - our intent is to draw attention to potential GTFS errors with stops far from the trip shape, and if we flag one that's a few meters off it shouldn't be problematic.